internal/quickfort/dialog.lua

```diff
@@ -202,7 +207,8 @@ local function dialog_command(command, text)
 end
 
 function do_dialog()
-    local blueprint_dialog = BlueprintDialog{
+    if blueprint_dialog then qerror('Quickfort gui dialog already open') end
```

Comment:
It may be worth having this dismiss the existing dialog instead of raising an error - in the (unlikely) event that the dialog gets into a broken state and doesn't respond to `Esc`, users that don't know about `devel/pop-screen` might try running `quickfort gui` again to reset it. (`gui/dfstatus` takes a slightly different approach and dismisses its dialog when run a second time, but I don't think that makes as much sense here.)

Reply:
This is a good idea. Thanks! I'll test, but I don't think the dismissed viewscreen on the stack is going to be a problem. I always call getViewscreen(true) to skip dismissed screens, and the df ui should work just fine with dismissed screens on the stack.

Reply:
Yeah, from a closer look at the logic, it looks like you're already feeding keys immediately after the dialog's `dismiss()` is called, so it would likely still be on the stack, and a second dismissed dialog on the stack shouldn't make a difference.